en/docs/tutorials/map-data.md

```diff
@@ -0,0 +1,293 @@
+
+#### Basic Mapping
+
+To map the person ID with the student ID, select **id** under **personDetails**, and then select **id** under **studentDetails** as shown below:
+
+![Basic mapping](../assets/img/tutorials/data-mapper/basic-mapping.gif){.cInlineImage-threeQuarters}
+
+#### Diagnose and fix mapping errors
+
+If any error occurs when you map parameters, you can fix them in the Statement Editor. To try it, follow the steps given below:
+
+1. Select **age** under **personDetails**, and then select **age** under **studentDetails**.
+
+    ![Match parameters with different data types](../assets/img/tutorials/data-mapper/match-parameters-with-different-data-types.gif){.cInlineImage-threeQuarters}
+
+    The connection appears in red to indicate an error in the data mapping.
+
+2. Click on the connection, move the cursor over the warning icon to view the error, and then click **Fix by editing expression**.
+
+    ![View error in Statement Editor](../assets/img/tutorials/data-mapper/view-error-in-statement-editor.gif){.cInlineImage-threeQuarters}
+
+3. Leave the expression selected (as by default), and in the **Suggestions** tab of the Statement Editor, click **toString()**.
+
+      Doing so converts the parameter value to the string format.
+
+      ![Change format](../assets/img/tutorials/data-mapper/change-format.gif){.cInlineImage-half}
+
+4. Click **Save**.
+
+The connection appears in blue to indicate that there are no errors.
+
+#### Aggregate multiple input fields to one output field
+
+You can derive one output parameter by combining two or more input parameters. In this example, the value for the `fullName` output parameter is a combination the values for the `firstName` and `lastName` input parameters.  You can map them as shown below:
+
+![Aggregate multiple input parameter](../assets/img/tutorials/data-mapper/aggregate-multiple-input-fields.gif){.cInlineImage-threeQuarters}
+   
+!!! info
+    By default, the Web Editor combines the two input parameters with a plus (**+**) operator. If you want to use a different operator or method to integrate two fields, click on the code icon edit the expression via the Statement Editor.<br/><br/>![Change operator](../assets/img/tutorials/data-mapper/change-operator.gif){.cInlineImage-threeQuarters}
+
+#### Map the arrays
+
+When the input and the output contain arrays, you can map an input array with the required output array and then convert the mapping to a query to map the individual parameters within the arrays.
+
+In the example in this tutorial, you need to map the `course` array under **courseDetails** with the `courses` array under **studentDetails**. Let's map them as follows:
+
+1. Select **course** in the **courseDetails** and then select **courses** under **studentDetails**.
+
+    ![Map arrays](../assets/img/tutorials/data-mapper/map-arrays.gif){.cInlineImage-threeQuarters}
+
+    The mapping appears in red because the types of the two record you selected are incompatible.
```

Comment:
Shall we say let's fix it at the end, because the user is given an error but not told we will fix it.

Reply:
Done